modules/consensus/synchronize_test.go

```diff
@@ -225,3 +227,87 @@ func TestBlockHistory(t *testing.T) {
 		}
 	}
 }
+
+// mockGatewayCountBroadcasts implements modules.Gateway to mock the Broadcast
+// method.
+type mockGatewayCountBroadcasts struct {
+	modules.Gateway
+	numBroadcasts int
+	mu            sync.RWMutex
+}
+
+// Broadcast is a mock implementation of modules.Gateway.Broadcast that
+// increments a counter denoting the number of times it's been called.
+func (g *mockGatewayCountBroadcasts) Broadcast(string, interface{}) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	g.numBroadcasts++
+	return
+}
+
+// TestSendBlocksBroadcastsOnce tests that the SendBlocks RPC call only
+// Broadcasts one block, no matter how many blocks are sent. In the case 0
+// blocks are sent, tests that Broadcast is never called.
+func TestSendBlocksBroadcastsOnce(t *testing.T) {
+	// Setup consensus sets.
+	cst1, err := blankConsensusSetTester("TestSendBlocksBroadcastsOnce1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst1.Close()
+	cst2, err := blankConsensusSetTester("TestSendBlocksBroadcastsOnce2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst2.Close()
+	// Setup mock gateway.
+	mg := mockGatewayCountBroadcasts{Gateway: cst1.cs.gateway}
+	cst1.cs.gateway = &mg
+	err = cst1.cs.gateway.Connect(cst2.cs.gateway.Address())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		blocksToMine          int
+		expectedNumBroadcasts int
+	}{
+		{0, 0},
+		{1, 1},
+		{2, 1},
+		{int(MaxCatchUpBlocks), 1},
+		{2 * int(MaxCatchUpBlocks), 1},
+		{2*int(MaxCatchUpBlocks) + 1, 1},
```

Comment:
they don't! Try it yourself. `go vet` only complains about unkeyed fields on imported types.